src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11057,67 +11099,28 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
                 }
                 else
                 {
-                    break;
+                    // Something that doesn't expand the current expression we're looking at.  Bail out and see if we
+                    // can end with a conditional expression.
+                    return false;
                 }
 
                 var newPrecedence = GetPrecedence(opKind);
 
-                // check for >>, >>=, >>> or >>>=
-                int tokensToCombine = 1;
```

Comment:
also a difficult value to reason about, with complex semantics about hwo it gets set.  no need for it at all.  We have opKind.  So we know precisely what tokens we have and if we have to merge them.